src/mctp-req.c

```diff
@@ -98,33 +103,26 @@ static int mctp_req(unsigned int net, mctp_eid_t eid,
 				sizeof(struct sockaddr_mctp));
 
 
-	printf("req:  message from (net %d, eid %d) type %d len %zd: 0x%02x..\n",
+	printf("req:  message from (net %d, eid %d) type %d len %zd\n",
 			addr.smctp_base.smctp_network, addr.smctp_base.smctp_addr.s_addr,
-			addr.smctp_base.smctp_type,
-			len,
-			rxbuf[0]);
+			addr.smctp_base.smctp_type, len);
 	if (addrlen == sizeof(struct sockaddr_mctp_ext)) {
 		printf("      ext ifindex %d ha[0]=0x%02x len %hhu\n",
 			addr.smctp_ifindex,
 			addr.smctp_haddr[0], addr.smctp_halen);
 	}
 
-	for (i = 0; i < len; i++) {
-		uint8_t exp = data ? data[i] : i & 0xff;
-		if (rxbuf[i] != exp)
-			errx(EXIT_FAILURE,
-				"payload mismatch at byte 0x%zx; "
-					"sent 0x%02x, received 0x%02x",
-				i, exp, rxbuf[i]);
-	}
-
+	printf("data:\n");
+	for (i = 0; i < len; i++)
+		printf("0x%02x ", rxbuf[i]);
```

Comment:
I'd suggest printing without the `0x` prefix.

We also have the `mctp_hexdump` helper available, which would give a proper hexdump-style output.

Reply:
Good to know, use `mctp_hexdump` instead.